apps/examples/kernel_sample/kernel_sample_main.c

```diff
@@ -234,19 +171,22 @@ static int user_main(int argc, char *argv[])
 #endif
 {
 	int i;
+#ifndef CONFIG_DISABLE_SIGNALS
+	struct mallinfo mmbefore;
+	struct mallinfo mmafter;
+#endif
 
 	/* Sample the memory usage now */
 
 #ifndef CONFIG_DISABLE_SIGNALS
 	usleep(HALF_SECOND_USEC);
 
 #ifdef CONFIG_CAN_PASS_STRUCTS
-	g_mmbefore = mallinfo();
-	g_mmprevious = g_mmbefore;
+	mmbefore = mallinfo();
 #else
-	(void)mallinfo(&g_mmbefore);
-	memcpy(&g_mmprevious, &g_mmbefore, sizeof(struct mallinfo));
+	(void)mallinfo(&mmbefore);
 #endif
+	set_mmprevious(&mmbefore);
```

Comment:
Let's macro this and after as well
```
user main()
{
  check_initial_memory_usages();
  test_a();
  check_test_memory_usages();
  test_b();
   check_test_memory_usages();
  ...
  check_last_memory_usages();
   show_memory_usages()l
}

check_initial_memory_usages()
{
    mmstart =mallinfo();
}

check_last_memory_usages()
{
  mm_end = mallinfo();
}

show...()
{
printf ( end - star);
}
```
I think we can move all of mm stubs in the new file. and we don't need to give any args through the function.